### PR TITLE
Add REST catalog spec to iceberg-core.jar

### PR DESCRIPTION
**Pull request by snazy**
Having the OpenAPI spec file used by a specific iceberg-core jar helps a lot.
This enables generating code from the spec by using the spec in the iceberg-core jar, without having to pull it "manually" from GH.

### CODE REVIEW

**Comment by Fokko** [comment]
@snazy Out of curiosity, do you use the open api spec to directly generate code? In Python, we tried to do it as well, but the structure was too complex in the end, and we used the generated code as a start. For example, the types can be both a string (`timestamp`, `fixed[12]`, etc) and a JSON dictionary (`{"type": "map", ...}`).

**Comment by snazy** [comment]
I think it's useful for two purposes:
1. to have the spec together with the code, to know how the spec for the code look(ed|s) like
2. to be able to generate some code - at least as a starter

Types are "an issue", yes. Not sure how much it would affect generated code in general  / across all languages. Jackson for example handles a "generic `JsonNode`" (which can be "anything") perfectly fine - I imagine that it would work similar for a "generic `j.l.Object`".

**Comment by haizhou-zhao** [comment]
@snazy I ran into the same issue as @Fokko mentioned above (I summarized my problem here: https://github.com/apache/iceberg/issues/6798). In addition to @Fokko's question, I want to know if you are writing some sort of model converter to convert, for example, between the [openAPI generated Schema object](https://github.com/apache/iceberg/blob/master/open-api/rest-catalog-open-api.yaml#L1075) and the [original iceberg Schema object](https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/Schema.java#L51). If not, would it be possible to know better how you are using the open API generated models? Thank you!

**Comment by snazy** [comment]
I still think, that a OpenAPI specification, if it exists, should technically match the implementation and vice versa. But that's actually not the case at the moment and code generation is not possible (commented [here](https://github.com/apache/iceberg/pull/6701#issuecomment-1414201166)).

There is no good JAX-RS Java code generator out there, but the ones available for node and python seem to work reasonably well.

Basically all projects that have an OpenAPI spec do publish it at least along with their published artifacts and/or via platforms like swagger-hub, from which others can take it or inspect it.
